js/index.js

```diff
@@ -26,17 +43,31 @@ function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
   //... your code goes here
+  const productRow = target.parentNode.parentNode
+  const productTable = productRow.parentNode
+
+  productTable.removeChild(productRow)
+
+  calculateAll();
 }
 
 // ITERATION 5
 
 function createProduct() {
-  //... your code goes here
+  const createProductRowFoot = document.querySelector('tfoot')
+  const newProductNameInput = createProductRowFoot.querySelector('input').value
 }
 
```

Comment:
Here's a working solution for `createProduct`, let me know if you have any questions over it
```
function createProduct() {
  const createProductRow = document.querySelector(".create-product");
  const newProductNameInputValue = createProductRow.querySelector("input").value;

  const newProductPriceValue = createProductRow.querySelector("input[type='number']");
  const newProductPrice = Number(newProductPriceValue.value).toFixed(2);

  const newProductRow = document.createElement("tr") 
  newProductRow.className= "product";
   newProductRow.innerHTML = `<tr class="product">
    <td class="name"> <span>${newProductNameInputValue}</span></td>
    <td class="price">$<span>${newProductPrice}</span></td>
    <td class="quantity"> <input type="number" value="0" min="0" placeholder="Quantity" /></td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action"><button class="btn btn-remove">Remove</button></td>
   </tr>  `;

const productTableBody = document.querySelector("tbody");
productTableBody.appendChild(newProductRow);

const newProductRemoveButton = newProductRow.querySelector(".btn-remove");
newProductRemoveButton.addEventListener("click", removeProduct);
}